src/corvus.c

```diff
@@ -18,14 +18,33 @@
 #include "timer.h"
 #include "alloc.h"
 
+static pthread_spinlock_t signal_lock;
 static struct context *contexts;
 
-void sig_handler(int sig)
+void sigsegv_handler(int sig)
 {
-    int i;
+    if (pthread_spin_trylock(&signal_lock) == EBUSY) {
+        sigset_t mask;
+        sigemptyset(&mask);
+        pselect(0, NULL, NULL, NULL, NULL, &mask);
```

Comment:
> I wonder whether pselect can be interrupted by some other unexpected signals, then the codes below are not protected by spinlock.

@doyoubi Yes, `pselect` will return after other signal is handled.

In most cases this will work fine, since Corvus will exit quickly after the very first `SIGSEGV`.